src/node/rpc/serialization.h

```diff
@@ -111,9 +111,17 @@ namespace ccf
   DECLARE_JSON_REQUIRED_FIELDS(EndpointMetrics::Out, metrics)
 
   DECLARE_JSON_TYPE(GetReceipt::In)
-  DECLARE_JSON_REQUIRED_FIELDS(GetReceipt::In, commit)
+  DECLARE_JSON_REQUIRED_FIELDS(GetReceipt::In, transaction_id)
+  DECLARE_JSON_TYPE_WITH_OPTIONAL_FIELDS(GetReceipt::Node)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-parameter"
+#pragma clang diagnostic ignored "-Wgnu-zero-variadic-macro-arguments"
+  DECLARE_JSON_REQUIRED_FIELDS(GetReceipt::Node);
+#pragma clang diagnostic pop
+  DECLARE_JSON_OPTIONAL_FIELDS(GetReceipt::Node, left, right)
```

Comment:
More of a comment: I agree this should end up as a `oneOf` in OpenAPI. I'm not sure how we'd represent this in the macro, and adding support is not worth the effort at the moment. I think this is a great example of a type that's complicated enough it should have custom `to_json`+schema builder functions, but this is close enough for now.

Reply:
Agreed.